pkg/tests/fixtures.go

```diff
@@ -424,6 +424,68 @@ func NewServiceFixture(servicePorts ...v1.ServicePort) *v1.Service {
 	}
 }
 
+// NewEndpointsFixtureWithSameNameMultiplePorts makes a new endpoint for testing
+func NewEndpointsFixtureWithSameNameMultiplePorts() *v1.Endpoints {
+	return &v1.Endpoints{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ServiceName,
+			Namespace: Namespace,
+		},
+		Subsets: []v1.EndpointSubset{
+			{
+				// IP addresses which offer the related ports that are marked as ready. These endpoints
+				// should be considered safe for load balancers and clients to utilize.
+				// +optional
+				Addresses: []v1.EndpointAddress{
+					{
+						IP: "10.9.8.7",
+						// The Hostname of this endpoint
+						// +optional
+						Hostname: "www.contoso.com",
+						// Optional: Node hosting this endpoint. This can be used to determine endpoints local to a node.
+						// +optional
+						NodeName: to.StringPtr(NodeName),
+					},
+				},
+				// IP addresses which offer the related ports but are not currently marked as ready
+				// because they have not yet finished starting, have recently failed a readiness check,
+				// or have recently failed a liveness check.
+				// +optional
+				NotReadyAddresses: []v1.EndpointAddress{},
+				// Port numbers available on the related IP addresses.
+				// +optional
+				Ports: []v1.EndpointPort{
+					{
+						Protocol: v1.ProtocolTCP,
+						Name:     "--service-https-port--",
```

Comment:
we have the variable ServiceHTTPPort defined already, we can reuse it here as port Name